types/react-gallery-carousel/index.d.ts

```diff
@@ -111,35 +130,35 @@ declare namespace Carousel {
             transitionDurationMin: number;
             transitionDurationMax: number;
             widgetsHasShadow: boolean;
-            hasLeftButton: ButtonType;
-            hasRightButton: ButtonType;
-            hasMediaButton: ButtonType;
-            hasSizeButton: ButtonType;
-            hasIndexBoard: ButtonType;
-            hasDotButtons: ButtonType;
-            hasCaptions: ButtonType;
+            hasLeftButton: SmallWidgetPositions | boolean;
+            hasRightButton: SmallWidgetPositions | boolean;
+            hasMediaButton: SmallWidgetPositions | boolean;
+            hasSizeButton: SmallWidgetPositions | boolean;
+            hasIndexBoard: SmallWidgetPositions | boolean;
+            hasDotButtons: LargeWidgetPositions | boolean;
+            hasCaptions: LargeWidgetPositions | boolean;
             hasThumbnails: boolean;
-            hasLeftButtonAtMax: ButtonType;
-            hasRightButtonAtMax: ButtonType;
-            hasMediaButtonAtMax: ButtonType;
-            hasSizeButtonAtMax: ButtonType;
-            hasIndexBoardButtonAtMax: ButtonType;
-            hasDotButtonsAtMax: ButtonType;
-            hasCaptionsAtMax: ButtonType;
+            hasLeftButtonAtMax: SmallWidgetPositions | boolean;
+            hasRightButtonAtMax: SmallWidgetPositions | boolean;
+            hasMediaButtonAtMax: SmallWidgetPositions | boolean;
+            hasSizeButtonAtMax: SmallWidgetPositions | boolean;
+            hasIndexBoardButtonAtMax: SmallWidgetPositions | boolean;
+            hasDotButtonsAtMax: LargeWidgetPositions | boolean;
+            hasCaptionsAtMax: LargeWidgetPositions | boolean;
```

Comment:
A lot of these props now has the possible positions available specified there
https://github.com/yifaneye/react-gallery-carousel#props

It's either LargeWidthPositions or SmallWidgetPositions, with a boolean also as an alternative.